internal/object/test/object.service_test.go

```diff
@@ -28,4 +39,149 @@ func (t *ObjectServiceTest) SetupTest() {
 		BucketName: "mock-bucket",
 		Endpoint:   "mock-endpoint",
 	}
+	t.uploadObjectRequest = &proto.UploadObjectRequest{
+		Filename: "object",
+		Data:     []byte("data"),
+	}
+}
+
+func (t *ObjectServiceTest) TestUploadInternalError() {
+	repo := mock_object.NewMockRepository(t.controller)
+	repo.EXPECT().Upload(t.uploadObjectRequest.Data, t.conf.BucketName, gomock.Any()).Return("", "", fmt.Errorf("error"))
+
+	svc := object.NewService(repo, t.conf, t.logger, utils.NewRandomUtils())
+
+	expectedErr := status.Error(codes.Internal, constant.InternalServerErrorMessage).Error()
+
+	actual, err := svc.Upload(context.Background(), t.uploadObjectRequest)
+
+	t.Nil(actual)
+	t.EqualError(err, expectedErr)
+}
+
+func (t *ObjectServiceTest) TestUploadSuccess() {
+	repo := mock_object.NewMockRepository(t.controller)
+	repo.EXPECT().Upload(t.uploadObjectRequest.Data, t.conf.BucketName, gomock.Any()).Return("url", "key", nil)
+
+	svc := object.NewService(repo, t.conf, t.logger, utils.NewRandomUtils())
+
+	_, err := svc.Upload(context.Background(), t.uploadObjectRequest)
+
+	t.Nil(err)
+}
```

Comment:
every test case should handle if the `actual` result is equal to `expected` result too
by doing so, we should we can make sure that our function work properly and return no error
```suggestion
	expected := &proto.UploadObjectResponse{
		Object: &proto.Object{
			Url: "url",
			Key: "key",
		},
	}

	actual, err := svc.Upload(context.Background(), t.uploadObjectRequest)

	t.Nil(err)
	t.Equal(expected, actual)
```

Reply:
please apply for all testcase